src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ShardedReadsDatasource.java

```diff
@@ -0,0 +1,361 @@
+                Date started;
+
+                @Override
+                public void startBundle(Context c) throws Exception {
+                    super.startBundle(c);
+                    if (cloudStorageUrl) {
+                        storageClient = GCSOptions.Methods.createStorageClient(c.getPipelineOptions().as(GCSOptions.class), auth);
+                    }
+                    started = new Date();
+                    readsEmitted = 0;
+                    querying = Stopwatch.createUnstarted();
+                    loading = Stopwatch.createUnstarted();
+                    choosing = Stopwatch.createUnstarted();
+                    shipping = Stopwatch.createUnstarted();
+                }
+
+                @Override
+                public void finishBundle(Context c) throws Exception {
+                    super.finishBundle(c);
+                    final Instant instant = new Date().toInstant();
+                    long ms = Duration.between(started.toInstant(), instant).toMillis();
+                    String q = querying.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    String l = loading.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    String ch = choosing.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    String s = shipping.elapsed(TimeUnit.MILLISECONDS) + " ms";
+                    bunny.stepEnd("Bundle sent " + readsEmitted + " reads in " + ms + " ms. Breakdown: querying " + q + " loading " + l + " choosing " + ch + " shipping " + s);
+                }
+
+                @Override
+                public void processElement(ProcessContext c) throws Exception {
+                    int rejected = 0;
+                    int accepted = 0;
+                    Stopwatch processing = Stopwatch.createStarted();
+
+                    Stopwatch kving = Stopwatch.createUnstarted();
+                    Stopwatch outputting = Stopwatch.createUnstarted();
+
+                    Stopwatch readingInput = Stopwatch.createStarted();
+                    SimpleInterval i = c.element();
+                    readingInput.stop();
+                    Stopwatch init = Stopwatch.createStarted();
+                    ArrayList<SimpleInterval> is = new ArrayList<>();
+                    is.add(i);
+
+                    Stopwatch cutShards = Stopwatch.createStarted();
+                    List<SimpleInterval> shards = cutToShards(is, outputShardSize);
+                    cutShards.stop();
+
+                    int currentShardIndex = 0;
+                    SimpleInterval currentShard = shards.get(currentShardIndex);
+                    List<GATKRead> ret = new ArrayList<GATKRead>();
+                    init.stop();
+
+                    if (!cloudStorageUrl && !hadoopUrl) {
+                        // local file, use setIntervalsForTraversal
+                        querying.start();
+                        ReadsDataSource rds = new ReadsDataSource(new File(bam));
```

Comment:
Also, since you are only querying a single interval, it's better to use `query()` rather than `setIntervalsForTraversal()` followed by `iterator()`